packages/paste-core/components/popover/__tests__/index.spec.tsx

```diff
@@ -94,9 +101,11 @@ describe('Popover', () => {
       expect(renderedPopoverButton.getAttribute('aria-controls')).toEqual('test-id');
       expect(renderedPopoverButton.getAttribute('aria-expanded')).toEqual('false');
       expect(renderedPopover).not.toBeVisible();
-      renderedPopoverButton.click();
-      await waitFor(() => expect(renderedPopoverButton.getAttribute('aria-expanded')).toEqual('true'));
-      await waitFor(() => expect(renderedPopover).toBeVisible());
+      await waitFor(() => {
+        userEvent.click(renderedPopoverButton);
+      });
+      expect(renderedPopoverButton.getAttribute('aria-expanded')).toEqual('true');
+      expect(renderedPopover).toBeVisible();
```

Comment:
These might have been causing async test problems due to state updates, but I'm not sure they are necessary changes. 

`ButtonTwo.click();` will cause a state update but the update isn't wrapped in `act` which is what you need to prevent a async test problem. by being strict about use of `userEvent`, we at least guarantee the state update is wrapped by `act`, as `userEvent` does this for you.

Not convinced you need to `waitFor` subsequently, but it's hard to debug locally